### PR TITLE
Adding Directions API docs

### PR DESCRIPTION
**Pull request by langsmith**
The `directions-overview` branch is very outdated, so I created a new branch off of `mb-pages` just now. This PR is a work in progress and will add Android documentation about the Directions API.

Remaining sections to build out before final additions/editing:

- [ ] Directions profiles (traffic, driving, cycling, walking)
- [ ] Alternative routes
- [ ] Bearings and continue straight and how that effects the route
- [ ] Section on legs
- [ ] Section on steps
- [ ] Section on  #manuevers

### CODE REVIEW

**Comment by langsmith** [comment]
Wow, I finally got around to this beast of a page. I think this is ready for review @colleenmcginnis / @osana . 

As I mention 1-2 times on the page, so much has already been written at https://www.mapbox.com/api-documentation/#directions, that I don't think I need to re-hash the specifics about legs, steps, etc. If anyone disagrees, please let me know. I'm happy to chat about what Java/Android-specific things should be mentioned in this pr. Others might be more familiar with hidden gotchas, etc. 


screencapture-localhost-8080-android-docs-java-overview-directions-2018-11-29-16_57_07

**Comment by langsmith** [comment]
Resolves #8

**Comment by langsmith** [comment]
cc @mapbox/navigation-android as fyi about this pr in case there's anything you want specifically mentioned on this Directions API page. Perhaps specific information here will lessen the number of questions ya'll get about the Navigation SDK for Android....

**Comment by colleenmcginnis** [comment]
>As I mention 1-2 times on the page, so much has already been written at https://www.mapbox.com/api-documentation/#directions, that I don't think I need to re-hash the specifics about legs, steps, etc. If anyone disagrees, please let me know. 

@langsmith I'm going to tag in @HeyStenson to review instead of me since she's more familiar with the current content of https://www.mapbox.com/api-documentation/#directions.

**Comment by langsmith** [comment]
cc @osana as fyi that this is moving

File: src/pages/java/overview/directions.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+Feature directionsRouteFeature = Feature.fromGeometry(LineString.fromPolyline(currentRoute.geometry(), PRECISION_6));
+`}
+
+kotlin={`
+val directionsRouteFeature = Feature.fromGeometry(LineString.fromPolyline(currentRoute.geometry(), PRECISION_6))
+`}
+/>
+}}
+
+Once you've created a `Feature` whose geometry is a `LineString`, you can then use it with the `FeatureCollection`, `GeoJsonSource`, and `LineLayer` objects. This is called "data-driven styling" and [more information about it can be found here](/android-docs/maps/overview/data-driven-styling/).
```
Comment (HeyStenson) [inline src/pages/java/overview/directions.md]: Maybe reword a little: 

"a `Feature` whose geometry is a..." ➡️ "a `Feature`for which the geometry is a..."

"This is called "data-driven styling" and [more information about it can be found here](/android-docs/maps/overview/data-driven-styling/)." ➡️ "This is called _data-driven styling_. For more information about how data-driven styling can be used in a map, see the [Data-driven styling guide](/android-docs/maps/overview/data-driven-styling/)."

File: src/pages/java/overview/index.md [hunk 2]
```diff
@@ -59,26 +59,24 @@ Version _below_ 3.0.0:
...
 
-### Position and Point objects
+### Point object
 
-Throughout the APIs exposed within Mapbox-Java, you'll notice the Position object's used heavily representing a coordinate. The order in which the coordinate pair are longitude followed by latitude. Creating a new Position is fairly straightforward and can be converted from a LatLng object if you are using our Maps SDK.
+Throughout the APIs exposed within the Java SDK, you'll notice the common usage of the `Point` object to represent a coordinate. The order in which the coordinate pair are longitude followed by latitude. Creating a new `Point` is fairly straightforward and can be converted from a `LatLng` object if you are using the Mapbox Maps SDK for Android.
 
-The SDK will try to detect if your coordinates are outside the excepted range and log a warning. This range is between -180 to 180 for longitudes and -90 to 90 for latitude.
+The Java SDK will try to detect if your coordinates are outside the excepted range and log a warning. This range is between -180 to 180 for longitudes and -90 to 90 for latitude.
```
Comment (HeyStenson) [inline src/pages/java/overview/index.md]: "excepted range" == "accepted range"?

File: src/pages/java/overview/directions.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+  - "import ToggleableCodeBlock from '../../../components/toggleable-code-block';"
+  - "import { JAVA_SDK_VERSION } from '../../../constants';"
+
+---
+
+The Mapbox Directions API delivers routes for navigating the world. Driving, walking, and cycling directions are all possible for you to request, retrieve, and use in your Android project. The Mapbox Java SDK's `MapboxDirections` class eases interaction with the [Mapbox Directions API](https://www.mapbox.com/api-documentation/#directions).
+
+The Directions API does not return estimated times of arrival (ETA). Please see [the Mapbox Matrix API](/android-docs/mapbox-services/overview/directions-matrix/) if you would like ETAs for your Android project. Conversely, the Matrix API _does not_ provide directions routes and distances, which the Directions API _does_ provide.
+
+Read [here](https://www.mapbox.com/api-documentation/#directions) for more extensive documentation about the Mapbox Directions API.
```
[the remark this answers is not in the document]
Comment (langsmith) [reply src/pages/java/overview/directions.md]: Fixed

File: src/pages/java/overview/directions.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+
+The Directions API can be used completely on its own to get routes and other directional information. You should also know that [the Mapbox Navigation SDK for Android](https://www.mapbox.com/android-docs/navigation/overview/) is built "on top of" the Directions API. The Navigation SDK enables users to more easily interact with and use the information inside of the Directions API response. _Use the Mapbox Java Services dependency if you want to use the Mapbox Directions API without using the entire Mapbox Navigation SDK._
+
+```groovy
+implementation 'com.mapbox.mapboxsdk:mapbox-sdk-services:{{ JAVA_SDK_VERSION }}'
+```
+
+## Buiding a request
+
+The Mapbox Directions API requires you to build a URL with various parameters to request the information which you want. This URL has required and optional parameters. The `MapboxDirections.Builder()` class helps you set these various parameters.
```
[the remark this answers is not in the document]
Comment (langsmith) [reply src/pages/java/overview/directions.md]: fixed

File: src/pages/java/overview/directions.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+	.profile(DirectionsCriteria.PROFILE_DRIVING)
+	.accessToken(MAPBOX_ACCESS_TOKEN)
+	.build()
+`}
+/>
+}}
+
+**Profiles**:
+
+- *Traffic*: For automotive routing. This profile factors in current and historic traffic conditions to avoid slowdowns. Traffic information is available in [these supported geographies](https://www.mapbox.com/api-documentation/pages/traffic-countries.html) .
```
[the remark this answers is not in the document]
Comment (langsmith) [reply src/pages/java/overview/directions.md]: Removed

File: src/pages/java/overview/directions.md [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+```
+
+## Buiding a request
+
+The Mapbox Directions API requires you to build a URL with various parameters to request the information which you want. This URL has required and optional parameters. The `MapboxDirections.Builder()` class helps you set these various parameters.
+
+The three required parameters are:
+
+- an origin `Point` object
+- a destination `Point ` object
```
[the remark this answers is not in the document]
Comment (langsmith) [reply src/pages/java/overview/directions.md]: Fixed

File: src/pages/java/overview/directions.md [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+
+
+[Here's full information on all of the optional parameters](https://www.mapbox.com/api-documentation/#retrieve-directions).
+
+Directions API requests for driving, walking, and cycling routes can specify up to 25 total waypoints along the route.  
+
+Requests using the traffic profile can specify up to 3 waypoints.
+
+- Traffic coverage for the traffic profile is available in supported geographies. Requests to this profile revert to driving profile results for areas without traffic coverage.
+- Maximum 60 requests per minute
```
[the remark this answers is not in the document]
Comment (langsmith) [reply src/pages/java/overview/directions.md]: Added
